app/src/org/commcare/models/database/global/GlobalDatabaseUpgrader.java

```diff
@@ -124,7 +132,44 @@ private boolean upgradeFourFive(SQLiteDatabase db) {
     }
 
     private boolean upgradeFiveSix(SQLiteDatabase db) {
-        return addTableForNewModel(db, ConnectKeyRecord.STORAGE_KEY, new ConnectKeyRecord());
+        return addTableForNewModel(db, ConnectKeyRecord.STORAGE_KEY, new ConnectKeyRecordV6());
+    }
+
+    private boolean upgradeSixSeven(SQLiteDatabase db) {
+        db.beginTransaction();
+
+        //Migrate the old ConnectKeyRecord in storage to the new version including isLocal
+        try {
+            db.execSQL(DbUtil.addColumnToTable(
+                    ConnectKeyRecord.STORAGE_KEY,
+                    ConnectKeyRecord.IS_LOCAL,
+                    "TEXT"));
+
+            SqlStorage<Persistable> oldStorage = new SqlStorage<>(
+                    ConnectKeyRecordV6.STORAGE_KEY,
+                    ConnectKeyRecordV6.class,
+                    new ConcreteAndroidDbHelper(c, db));
+
+            SqlStorage<Persistable> newStorage = new SqlStorage<>(
+                    ConnectKeyRecord.STORAGE_KEY,
+                    ConnectKeyRecord.class,
+                    new ConcreteAndroidDbHelper(c, db));
+
+            for (Persistable r : oldStorage) {
+                ConnectKeyRecordV6 oldRecord = (ConnectKeyRecordV6)r;
+                ConnectKeyRecord newRecord = ConnectKeyRecord.fromV6(oldRecord);
+                //set this new record to have same ID as the old one
+                newRecord.setID(oldRecord.getID());
+                newStorage.write(newRecord);
+            }
+
+            db.setTransactionSuccessful();
+            return true;
+        } catch(Exception e) {
+            return false;
```

Comment:
think we should not be catching exceptions here.

Reply:
True. @OrangeAndGreen any particular reason for not letting this bubble up? We might not be able to pick this up if not.

Reply:
Think it was a mistake and Dave is planning to remove it from Connect branch as well

Reply:
Yeah, Shubham is correct, we shouldn't catch the exception and should let it bubble up instead.